root/artist/header.tt

```diff
@@ -1,4 +1,4 @@
-<div class="artistheader">
+<div class="artistheader [% artist.type_name.lower() _ '-icon' IF artist.type.name %]">
```

Comment:
Two aesthetic issues:
- When there is no type, this will result in `class="artistheader "`. While that works fine, the space is a bit annoying. You could change it into `[%~ ' ' _ artist....` – the tilde will cause TT to drop all preceding whitespace, and a space will be output together with the type name if it exists.
- While TT VMethods without parameters optionally can have an empty pair of parentheses, it is more common to omit it. So `.lower()` should preferably be just `.lower`.